### PR TITLE
#4282 Fix TSX lang install error

### PR DESCRIPTION
**Pull request by Dax911**
more clearly separated args

Release Notes:

- Fixed install issue with TSX language server ([#4282](https://github.com/zed-industries/zed/issues/4282)).

### CODE REVIEW

**Comment by maxdeviant** [comment]
@Dax911 Could you explain what the fix is here (and also what the problem is that it is fixing)?

I wouldn't expect that passing a `&PathBuf` to `arg` would make any different over passing a `PathBuf` directly, since `Command::arg` takes an `AsRef<OsStr>`: https://docs.rs/async-process/latest/async_process/struct.Command.html#method.arg

**Comment by Dax911** [comment]
> @Dax911 Could you explain what the fix is here (and also what the problem is that it is fixing)?
> 
> I wouldn't expect that passing a `&PathBuf` to `arg` would make any difference over passing a `PathBuf` directly, since `Command::arg` takes an `AsRef<OsStr>`: https://docs.rs/async-process/latest/async_process/struct.Command.html#method.arg

The Command::arg method expects a type that implements AsRef<OsStr>, and &PathBuf satisfies this requirement. PathBuf is a more complex type that represents an owned, growable path while &Path is a reference to a path. When you pass &PathBuf to Command::arg, it gets converted to &Path, and the underlying OsStr representation of the path is used as a separate argument in the command.

New to rust but I was just reading about the PathBuf, hmm I see the attached resource from the docs and I think I am picking up what you are putting down. I will fully admit that I could totally be barking up the wrong tree in fixing this bug making that assumption w `nnpm` being introduced here. But I do think this would be better rust code anyway? Would love to get some feedback from more rust devs. Hmm

**Comment by Dax911** [comment]
I am dumb I tried to be smarter than rust... I should know better than that. Reread the info u attached. In the context of the async-process crate, the `Command::arg` method indeedd takes any type that implements `AsRef<OsStr>`. Both PathBuf and `&PathBuf` satisfy this trait. So, passing either `npm_file` or `&npm_file` as an argument to arg should theoretically work without any issue.

Given this, the original suggestion to change from `command.arg(npm_file)` to `command.arg(&npm_file)` is prob not directly related to this bug.

TLDR; My bad. Skill issue.

**Comment by maxdeviant** [comment]
> But I do think this would be better rust code anyway? Would love to get some feedback from more rust devs. Hmm

There isn't really any advantage to passing a ref here vs passing the entire `PathBuf`.

The code we're calling calls `.as_ref()` on it either way, and since we don't need to use `npm_file` again it's fine for us to hand over ownership to `.arg`.

Appreciate your effort in opening the PR, but I'm going to close this one since it doesn't seem related to the issue in #4282.
